File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        int ans = -1;
        int streak = 0, left = 0, right = 0;
        while(right < num.length()) {
            if(num.charAt(left) == num.charAt(right)) {
                streak++;
                right++;
                if(streak >= 3) ans = Math.max(ans, num.charAt(left)-'0');
            }
            else {
                streak = 0;
                left = right;
            }
        }
        return ans == -1 ? "" : String.valueOf(ans) + String.valueOf(ans) + String.valueOf(ans);
    }
}

File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.js
/**
 * @param {string} num
 * @return {string}
 */
var largestGoodInteger = function(num) {
    let ans = -1;
    for(let i=0; i<num.length-2; i++) {
        if(num[i] === num[i+1] && num[i] === num[i+2]) {
            ans = Math.max(ans, num[i]);
        }
    }
    return ans === -1 ? "" : "" + ans + ans + ans;
};